core/src/forwarding_stage.rs

```diff
@@ -0,0 +1,618 @@
+use {
```

Comment:
I would love to see a bit more documentation at least for the new components. In particular:
* on the module level, what it is for overall
* on the struct/method level to describe the purpose, possible errors, etc

Reply:
d4799c84b4988a0a9c87c1c3cb4299a1e64353ef